packages/react-checkbox/src/components/Checkbox/Checkbox.base.tsx

```diff
@@ -6,130 +6,132 @@ import { Icon } from 'office-ui-fabric-react/lib/Icon';
 
 const getClassNames = classNamesFunction<ICheckboxStyleProps, ICheckboxStyles>();
 
-export const CheckboxBase = React.forwardRef((props: ICheckboxProps, forwardedRef: React.Ref<HTMLDivElement>) => {
-  const {
-    disabled,
-    inputProps,
-    name,
-    ariaLabel,
-    ariaLabelledBy,
-    ariaDescribedBy,
-    ariaPositionInSet,
-    ariaSetSize,
-    title,
-    label,
-    checkmarkIconProps,
-    styles,
-    theme,
-    className,
-    boxSide = 'start',
-  } = props;
-
-  const id = useId('checkbox-', props.id);
-
-  const rootRef = React.useRef<HTMLDivElement | null>(null);
-  const mergedRootRefs: React.Ref<HTMLDivElement> = useMergedRefs(rootRef, forwardedRef);
-  const inputRef = React.useRef<HTMLInputElement>(null);
-
-  const [isChecked, setIsChecked] = useControllableValue(props.checked, props.defaultChecked, props.onChange);
-  const [isIndeterminate, setIsIndeterminate] = useControllableValue(props.indeterminate, props.defaultIndeterminate);
-
-  useFocusRects(rootRef);
-  useDebugWarning(props);
-  useComponentRef(props, isChecked, isIndeterminate, inputRef);
-
-  const classNames = getClassNames(styles!, {
-    theme: theme!,
-    className,
-    disabled,
-    indeterminate: isIndeterminate,
-    checked: isChecked,
-    reversed: boxSide !== 'start',
-    isUsingCustomLabelRender: !!props.onRenderLabel,
-  });
-
-  const onChange = (ev: React.ChangeEvent<HTMLElement>): void => {
-    if (isIndeterminate) {
-      // If indeterminate, clicking the checkbox *only* removes the indeterminate state (or if
-      // controlled, lets the consumer know to change it by calling onChange). It doesn't
-      // change the checked state.
-      setIsChecked(!!isChecked, ev);
-      setIsIndeterminate(false);
-    } else {
-      setIsChecked(!isChecked, ev);
-    }
-  };
-
-  const defaultLabelRenderer = React.useCallback(
-    (checkboxProps?: ICheckboxProps): JSX.Element | null => {
-      if (!checkboxProps) {
-        return null;
-      }
-      return checkboxProps.label ? (
-        <span aria-hidden="true" className={classNames.text} title={checkboxProps.title}>
-          {checkboxProps.label}
-        </span>
-      ) : null;
-    },
-    [classNames.text],
-  );
-
-  const onRenderLabel = props.onRenderLabel || defaultLabelRenderer;
-
-  const ariaChecked: React.InputHTMLAttributes<HTMLInputElement>['aria-checked'] = isIndeterminate
-    ? 'mixed'
-    : isChecked
-    ? 'true'
-    : 'false';
-
-  const slotProps = {
-    root: {
-      className: classNames.root,
-      title,
-      ref: mergedRootRefs,
-    },
-    input: {
-      className: classNames.input,
-      type: 'checkbox' as React.InputHTMLAttributes<HTMLInputElement>['type'],
-      ...inputProps,
-      ref: inputRef,
-      checked: !!isChecked,
+export const CheckboxBase: React.FunctionComponent<ICheckboxProps> = React.forwardRef<HTMLDivElement, ICheckboxProps>(
```

Comment:
It might be worth adding a lint rule for this (as well as for component naming). The trick is how to figure out in a lint rule that something is a component file, but I have ideas for that, and we can always disable the rule in specific files where the heuristic gives wrong results. (We can make an issue to track this rather than doing it within the PR.)

Reply:
added an issue [here](https://github.com/microsoft/fluentui/issues/15048). pls update if i noted anything incorrectly